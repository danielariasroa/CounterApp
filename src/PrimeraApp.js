import React from 'react'
import PropTypes from 'prop-types'

//import React, {Fragment} from 'react'

//FC

const PrimeraApp = ({ saludo, subtitulo }) =>{

    //const saludo = 'Hola mundo';

    /*const saludo = {
            nombre: 'daniel',
            edad: 45 */   
    
    return(
       
        <>
            <h1> { saludo } </h1>
             {/*<pre> {JSON.stringify(saludo, null, 3)} </pre>*/}
             <h2>{subtitulo}</h2>
             <p> a la vivora vovora de la mar, por aqui pueden pasar</p>
        </>
   
    )

    }

    PrimeraApp.propTypes = {
        saludo: PropTypes.string.isRequired
    }
    PrimeraApp.defaultProps = {
        subtitulo: 'Soy un subtitulo'
    }
export default PrimeraApp;