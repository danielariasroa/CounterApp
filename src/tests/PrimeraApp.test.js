import React from 'react'
import '@testing-library/jest-dom'
import{ shallow } from 'enzyme';
import PrimeraApp from "../PrimeraApp";

describe('pruebas en primera app', ()=>{
    /*
test('Debe mostrar el mensaje de "Hola, soy daniel ', () => {
    const saludo = "Hola, soy daniel";

    const {getByText} = render(<PrimeraApp saludo={saludo}/>)

    expect(getByText(saludo)).toBeInTheDocument()
})
*/

test('debe de mostrar primera app correctamente ', () => {

    const saludo = 'Hola, soy daniel'
    const wrapper = shallow(<PrimeraApp saludo={saludo}/>)
    
    expect(wrapper).toMatchSnapshot();
})

test('debe mostrar el subtitulo enviado por props ', () => {
    
    const saludo = 'Hola, soy daniel';
    const subTitulo = 'Soy un subtitulo';

    const wrapper = shallow(
        <PrimeraApp
            saludo={saludo}
            subtitulo={subTitulo}
        
        />
    );

        const textoParrafo = wrapper.find('h2').text();

        expect(textoParrafo).toBe(subTitulo);

})


})