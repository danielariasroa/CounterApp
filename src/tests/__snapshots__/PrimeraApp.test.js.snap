// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en primera app debe de mostrar primera app correctamente  1`] = `
<Fragment>
  <h1>
     
    Hola, soy daniel
     
  </h1>
  <h2>
    Soy un subtitulo
  </h2>
  <p>
     a la vivora vovora de la mar, por aqui pueden pasar
  </p>
</Fragment>
`;
